R/db-odbc-mssql.R

```diff
@@ -223,11 +209,17 @@ mssql_generic_infix <- function(if_select, if_filter) {
 }
 
 mssql_sql_if <- function(cond, if_true, if_false = NULL) {
+
+  old <- sql_current_context()
```

Comment:
I think these you should the `on.exit()` pattern here:

```R
old <- sql_current_context()
on.exit(sql_current_context(old), add = TRUE)
```

But I don't understand how this code works because I don't see you checking that the context is "ifelse" anywhere.

Reply:
Yes, this was to prevent current context from returning "SELECT" or "ORDER".  My initial thought was to invent an overriding value in case it needs to be used later for something else. But, while I was re-reading my code and trying to remember what I did, the "ifelse" value does cause confusion.  Replaced it with `list(clause = "")`.  This also solves @ablack3 's issue reported here: https://github.com/tidyverse/dbplyr/pull/103#issuecomment-431481889

Reply:
Sorry, I typed the pattern slightly incorrectly - it should be:

```R
old <- set_current_context(list(clause = ""))
on.exit(sql_current_context(old), add = TRUE)
```

If that doesn't work `set_current_context()` needs to be modified to invisibly return the previous value.

Reply:
This still isn't quite right - you have three lines of code instead of two.